lottie/src/main/java/com/airbnb/lottie/animation/content/FillContent.java

```diff
@@ -120,6 +121,12 @@ public FillContent(final LottieDrawable lottieDrawable, BaseLayer layer, ShapeFi
       dropShadowAnimation.applyTo(paint);
     }
 
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+      final BlendModeCompat blendMode = layer.getBlendMode().toNativeBlendMode();
+      if (blendMode != null)
+        PaintCompat.setBlendMode(paint, blendMode);
```

Comment:
The entirety of setBlendMode wasn't added until API 29 so it's not just a matter of only supporting a subset of backwards-compatible ones, unfortunately.

Reply:
yeah I'm using [PaintCompat](https://developer.android.com/reference/androidx/core/graphics/PaintCompat) here which seems to offer up the ability to use [BlendModeCompat](https://developer.android.com/reference/androidx/core/graphics/BlendModeCompat). If the docs are up to spec, it should be able to give us anything that maps to the following in a backwards compatible way
   ```
  CLEAR,
  SRC,
  DST,
  SRC_OVER,
  DST_OVER,
  SRC_IN,
  DST_IN,
  SRC_OUT,
  DST_OUT,
  SRC_ATOP,
  DST_ATOP,
  XOR,
  PLUS,
  MODULATE,
  SCREEN,
  OVERLAY,
  DARKEN,
  LIGHTEN
  ```
  
  The implementation of `PaintCompat.setBlendMode()` looks like this
  ```java
public static boolean setBlendMode(@NonNull Paint paint, @Nullable BlendModeCompat blendMode) {
    if (Build.VERSION.SDK_INT >= 29) {
        Object blendModePlatform = blendMode != null
                ? BlendModeUtils.Api29Impl.obtainBlendModeFromCompat(blendMode) : null;
        Api29Impl.setBlendMode(paint, blendModePlatform);
        // All blend modes supported in Q
        return true;
    } else if (blendMode != null) {
        PorterDuff.Mode mode = obtainPorterDuffFromCompat(blendMode);
        paint.setXfermode(mode != null ? new PorterDuffXfermode(mode) : null);
        // If the BlendMode has an equivalent PorterDuff mode, return true,
        // otherwise return false
        return mode != null;
    } else {
        // Configuration of a null BlendMode falls back to the default which is supported in
        // all platform levels
        paint.setXfermode(null);
        return true;
    }
}
 ````

Reply:
Unfortunately that isn't a lot, but it's better than nothing imho,
It would give us (from lottie blend types) the following,

`NORMAL`, `SCREEN`, `OVERLAY`, `DARKEN`, `LIGHTEN`, and `ADD`

6/18 blend modes

Reply:
Unfortunately, it doesn't look like there is consistent behavior even for some of those.
1. If you look at the API 23 vs 31 diffs on the main PR page, there are several improved diffs for 31 but not for 23
2. If you look at the ones you added, if I'm reading this right, SCREEN should be bm2
31:

![31](https://s3.amazonaws.com/lottie-happo/snapshots/1f41346090f9374ed60aad310226fece.png)

23:

![23](https://s3.amazonaws.com/lottie-happo/snapshots/a1668279748dfddb7e742165f4b2ea0b.png)